File: Vectors , ArrayList, List/Python/KadaneAlgorithm.py
# Find the Sum of Maximum Sum Subarray.{Using Kadane's Algorithm}
# Time Complexity: O(n)
# Space Complexity: O(1)

list = [-2, -3, 4, -1, -2, 1, 5, -3]
max_sum = list[0]
current_sum = list[0]
for val in range(1, len(list)):
    current_sum += list[val]
    max_sum = max(max_sum, current_sum)
    if current_sum < 0:
        current_sum = 0
print(max_sum)

File: Vectors , ArrayList, List/C++/KadaneAlgorithm.cpp
// Find the Sum of Maximum Sum Subarray.{Using Kadane's Algorithm}
// Time Complexity: O(n)
// Space Complexity: O(1)

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int Kadane_Alogirithm(vector<int> &arr){
    int n = arr.size();
    int maxSum = arr[0];
    int currSum = arr[0];
    for (int i = 1; i < n; i++){
        currSum = max(arr[i], currSum + arr[i]);
        maxSum = max(maxSum, currSum);
    }
    return maxSum;
}

int main(){
    vector<int> arr = {-2, -3, 4, -1, -2, 1, 5, -3};
    cout << "Maximum Sum of Subarray: " << Kadane_Alogirithm(arr) << endl;
    return 0;
}